src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2175,6 +2201,25 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
 
             Debug.Assert(false, "Missing case in lambda conversion error reporting");
             diagnostics.Add(ErrorCode.ERR_InternalError, syntax.Location);
+
+            static void reportDefaultParameterMismatchError(BindingDiagnosticBag diagnostics, Location lambdaParameterLocation, int paramIdx, ConstantValue lambdaDefault, ConstantValue? delegateDefault)
+            {
+                if (lambdaDefault.IsBad || (delegateDefault?.IsBad ?? false))
+                {
+                    return;
+                }
+
+                var lambdaDefaultString = lambdaDefault.GetValueToDisplay() ?? "null";
```

Comment:
We do not want to call this early, it should only called by rendering the diagnostic. What problem were you trying to solve?

Reply:
`ConstantValue`s aren't formattable, so there will be an exception thrown if one is passed as part of a diagnostic. The workaround is to convert the underlying value into a string

Reply:
The workaround is to implement the needed interface on `ConstantValue` 🙂